lib/node-detector.js

```diff
@@ -37,6 +40,24 @@ class NodeDetector {
     }
   }
 
+  static async retrieveUsingWhereCommand () {
+    let stdout;
+    try {
+      stdout = (await exec('where', ['node'])).stdout;
+    } catch (ign) {
+      log.debug(ign);
+      return null;
+    }
+    let nodePath = stdout.replace("\n", "");
+    if (await fs.exists(nodePath)) {
+      log.debug(`Node binary found using where command at: ${nodePath}`);
+      return nodePath;
+    } else {
+      log.debug('Node binary not found using the where command.');
+      return null;
+    }
+  }
+
   static async retrieveUsingAppleScript () {
```

Comment:
This method, then, should only be called on Mac systems (there not existing AppleScript anywhere else).